schema/crdb/dbinit.sql

```diff
@@ -388,6 +388,11 @@ CREATE TABLE IF NOT EXISTS omicron.public.zpool (
     total_size INT NOT NULL
 );
 
+/* Create an index on the physical disk id */
+CREATE INDEX IF NOT EXISTS lookup_zpool_by_disk on omicron.public.zpool (
+    physical_disk_id
```

Comment:
This doesn't quite match up1.sql.  (Did this not cause a test failure?  If it didn't, that's a bug in the test, too.)

Reply:
Is the difference because `dbinit.sql` has:
```
CREATE INDEX IF NOT EXISTS lookup_zpool_by_disk on omicron.public.zpool (        
    physical_disk_id                                                             
) WHERE physical_disk_id IS NOT NULL AND time_deleted IS NULL;
```
And up1.sql has:
```
CREATE INDEX IF NOT EXISTS lookup_zpool_by_disk ON omicron.public.zpool (physical_disk_id, id) WHERE time_deleted IS NULL; 
```
I.e. the up1.sql does not have the `physical_disk_id IS NOT NULL` part?

The only test failures I see are:
` bootstore schemes::v0::peer::tests::network_config`
And that looks like it's related to the address already being in use (I'm on atrium, so it would not surprise me).

Where are the tests that you expected to fail so I can go figure out why it did not notice this difference.

Reply:
For context, this test involves applying updates and then comparing https://www.cockroachlabs.com/docs/stable/information-schema values.

However, we don't compare all those tables, just some of them. This may be a case we should consider adding.

Reply:
Filed: https://github.com/oxidecomputer/omicron/issues/4143

Reply:
@leftwo It's both the list of columns and the `WHERE` clause that differs between dbinit.sql and the up.sql files.

Reply:
I have fixes for both in https://github.com/oxidecomputer/omicron/pull/4152 - would appreciate a look whenever y'all get a chance!